x-pack/plugin/index-lifecycle/src/main/java/org/elasticsearch/xpack/indexlifecycle/IndexLifecycleRunner.java

```diff
@@ -150,7 +150,7 @@ static ClusterState moveClusterStateToNextStep(Index index, ClusterState cluster
             LongSupplier nowSupplier) {
         IndexMetaData idxMeta = clusterState.getMetaData().index(index);
         Settings.Builder indexSettings = moveIndexSettingsToNextStep(idxMeta.getSettings(), currentStep, nextStep, nowSupplier);
-        ClusterState.Builder newClusterStateBuilder = newClusterStateWithIndexSettings(index, clusterState, indexSettings);
+        ClusterState.Builder newClusterStateBuilder = newClusterStateWithIndexSettings(idxMeta.getIndex(), clusterState, indexSettings);
```

Comment:
Im curious: Why does this need to change?

Reply:
sorry, leftover from another route taken